x/ccv/provider/keeper/keeper.go

```
@@ -728,101 +742,79 @@ func (k Keeper) BondDenom(ctx sdk.Context) (string, error) {
-	for _, proposedChain := range proposedChains {
-		allConsumerChains = append(allConsumerChains, proposedChain.ChainID)
+// GetAllActiveConsumerIds returns all the consumer ids of chains that are registered, initialized, or launched
+func (k Keeper) GetAllActiveConsumerIds(ctx sdk.Context) []string {
+	latestConsumerId, found := k.GetConsumerId(ctx)
+	if !found {
+		return []string{}
 	}
-	pendingChainIDs := k.GetAllPendingConsumerChainIDs(ctx)
-	allConsumerChains = append(allConsumerChains, pendingChainIDs...)
 
-	return allConsumerChains
-}
-
-// SetTopN stores the N value associated to chain with `chainID`
-func (k Keeper) SetTopN(
-	ctx sdk.Context,
-	chainID string,
-	N uint32,
-) {
-	store := ctx.KVStore(k.storeKey)
-
-	buf := make([]byte, 4)
-	binary.BigEndian.PutUint32(buf, N)
-
-	store.Set(types.TopNKey(chainID), buf)
-}
+	consumerIds := []string{}
+	for i := uint64(0); i <= latestConsumerId; i++ {
+		consumerId := fmt.Sprintf("%d", i)
+		phase, foundPhase := k.GetConsumerPhase(ctx, consumerId)
+		if !foundPhase || (phase != Registered && phase != Initialized && phase != Launched) {
+			continue
+		}
+		consumerIds = append(consumerIds, consumerId)
+	}
 
-// DeleteTopN removes the N value associated to chain with `chainID`
-func (k Keeper) DeleteTopN(
-	ctx sdk.Context,
-	chainID string,
-) {
-	store := ctx.KVStore(k.storeKey)
-	store.Delete(types.TopNKey(chainID))
+	return consumerIds
 }
 
-// GetTopN returns (N, true) if chain `chainID` has a top N associated, and (0, false) otherwise.
+// GetTopN returns N if chain `consumerId` has a top N associated, and 0 otherwise.
 func (k Keeper) GetTopN(
 	ctx sdk.Context,
-	chainID string,
-) (uint32, bool) {
-	store := ctx.KVStore(k.storeKey)
-	buf := store.Get(types.TopNKey(chainID))
-	if buf == nil {
-		return 0, false
-	}
-	return binary.BigEndian.Uint32(buf), true
+	consumerId string,
+) uint32 {
+	powerShapingParameters, _ := k.GetConsumerPowerShapingParameters(ctx, consumerId)
```

Do not ignore errors.